internal/tool/generate/generator.go

```diff
@@ -98,10 +98,11 @@ func (list ErrorList) Error() string {
 func Generate(dir string, pkgs []string) error {
 	fset := token.NewFileSet()
 	cfg := &packages.Config{
-		Mode:      packages.NeedName | packages.NeedSyntax | packages.NeedImports | packages.NeedTypes | packages.NeedTypesInfo,
-		Dir:       dir,
-		Fset:      fset,
-		ParseFile: parseNonWeaverGenFile,
+		Mode:       packages.NeedName | packages.NeedSyntax | packages.NeedImports | packages.NeedTypes | packages.NeedTypesInfo,
+		Dir:        dir,
+		Fset:       fset,
+		ParseFile:  parseNonWeaverGenFile,
```

Comment:
It turns out that `packages.Load()` ends up compiling `weaver_gen.go` files before `parseNonWeaverGenFile()` is invoked.  So although we parse malformed `weaver_gen.go` files successfully, the overall package loading fails [here](https://github.com/ServiceWeaver/weaver/blob/9797c7f2923d966105e94e23dd4ed12718ad8fbb/internal/tool/generate/generator.go#L171-L176).

Note that the compilation of `weaver_gen.go` files happens as part of a `go list` command issued by `packages.Load()`.  Here is a sample invocation:

`go list -e -json=Name,ImportPath,Error,Dir,GoFiles,IgnoredGoFiles,IgnoredOtherFiles,CFiles,CgoFiles,CXXFiles,MFiles,HFiles,FFiles,SFiles,SwigFiles,SwigCXXFiles,SysoFiles,CompiledGoFiles,DepOnly,Imports,ImportMap,Export -compiled=true -test=false -export=true -deps=true -find=false`

We still need `parseNonWeaverGenFile()` since the build tag change only prevents `go list` from compiling `weaver_gen.go` files, but these files are still parsed as part of the [refine call](https://cs.opensource.google/go/x/tools/+/refs/tags/v0.8.0:go/packages/packages.go;drc=edddc5fc3223b7f12772f78079fcbcac3bd47d97;l=266) made by `packages.Load()`.

Reply:
It might be possible to avoid parsing `IgnoredGoFiles` in `refine()` but I haven't looked into that.

Reply:
Thank you for answering my question! I understand now